starwars_api/models.py

```diff
@@ -54,19 +60,38 @@ def __repr__(self):
 
 class BaseQuerySet(object):
 
-    def __init__(self):
-        pass
+    def __init__(self, subclass):
+        self.index = 0
+        self.max_iterations=0
+        self.subclass = subclass
+        self.get_data = getattr(api_client, 'get_'+ self.subclass.RESOURCE_NAME)
+        self.page = 1
 
     def __iter__(self):
-        pass
+        return self
 
     def __next__(self):
         """
         Must handle requests to next pages in SWAPI when objects in the current
         page were all consumed.
         """
-        pass
-
+        data = self.get_data("page="+str(self.page))
+        
+        if self.max_iterations != data["count"]:
+            obj = data["results"][self.index]
+            if self.subclass.RESOURCE_NAME=="people":
```

Comment:
This hardcoding is not good. Imagine if you want to add a third model in the future. You will need to remember updating this "if" statement to make it work. It would be better to do something dynamic using the camel case version of `self.RESOURCE_NAME`. `eval()` function could be useful.